homeassistant/components/stt/__init__.py

```diff
@@ -73,28 +224,71 @@ def __init__(self, providers: dict[str, Provider]) -> None:
 
     async def post(self, request: web.Request, provider: str) -> web.Response:
         """Convert Speech (audio) to text."""
-        if provider not in self.providers:
+        hass: HomeAssistant = request.app["hass"]
+        provider_entity: ProviderEntity | None = None
+        if (
+            not (provider_entity := async_get_provider_entity(hass, f"stt.{provider}"))
+            and provider not in self.providers
+        ):
             raise HTTPNotFound()
-        stt_provider: Provider = self.providers[provider]
 
         # Get metadata
         try:
             metadata = _metadata_from_header(request)
         except ValueError as err:
             raise HTTPBadRequest(text=str(err)) from err
 
-        # Check format
-        if not stt_provider.check_metadata(metadata):
-            raise HTTPUnsupportedMediaType()
+        if not provider_entity:
+            stt_provider: Provider | ProviderEntity = self.providers[provider]
 
-        # Process audio stream
-        result = await stt_provider.async_process_audio_stream(
-            metadata, request.content
-        )
+            if not self._legacy_provider_reported:
+                self._legacy_provider_reported = True
+                report_issue = self._suggest_report_issue(provider, stt_provider)
+                # This should raise in Home Assistant Core 2022.5
```

Comment:
```suggestion
```

Reply:
Instead or removing this comment, maybe push it forward by a few months to give custom components time to migrate?

Reply:
What do you think?
```suggestion
                # This should raise in Home Assistant Core 2023.9
```